libvmi/os/linux/core.c

```diff
@@ -233,24 +302,47 @@ static status_t init_from_rekall_profile(vmi_instance_t vmi)
     return ret;
 }
 
+/* Is this the page holding the init task? */
 static status_t init_task_kaslr_test(vmi_instance_t vmi, addr_t page_vaddr)
 {
     status_t ret = VMI_FAILURE;
     uint32_t pid;
+    addr_t addr;
     addr_t init_task = page_vaddr + (vmi->init_task & VMI_BIT_MASK(0,11));
     linux_instance_t linux_instance = vmi->os_data;
     access_context_t ctx = {
         .translate_mechanism = VMI_TM_PROCESS_DTB,
         .dtb = vmi->kpgd
     };
 
+    /* The pid should be 0 */
     ctx.addr = init_task + linux_instance->pid_offset;
     if ( VMI_FAILURE == vmi_read_32(vmi, &ctx, &pid) )
         return ret;
 
-    if ( pid )
+    if ( 0 != pid )
         return ret;
 
+    /* Kernel tasks have no mm */
+    ctx.addr = init_task + linux_instance->mm_offset;
+    if ( VMI_FAILURE == vmi_read_addr(vmi, &ctx, &addr) )
+        return ret;
+
+    if ( 0 != addr )
+        return ret;
+
+#if 0
+    /* Verify that the next task is readable */
```

Comment:
I plan to remove this check against task->tasks. I don't think it adds much value.